### PR TITLE
Fix spine debug draw only display in editor

### PR DESCRIPTION
**Pull request by yanjifa**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/spine/skeleton.ts [hunk 1]
```diff
@@ -227,10 +227,13 @@ export class Skeleton extends UIRenderer {
     @serializable
     protected _useTint = false;
     @serializable
+    @editorOnly
```
Comment (bofeng-song) [inline cocos/spine/skeleton.ts]: It's not a editorOnly property, it can be work on native and other platforms

File: cocos/spine/skeleton.ts [hunk 2]
```diff
@@ -227,10 +227,13 @@ export class Skeleton extends UIRenderer {
...
     protected _debugMesh = false;
     @serializable
+    @editorOnly
```
Comment (bofeng-song) [inline cocos/spine/skeleton.ts]: It's not a editorOnly property, it can be work on native and other platforms

File: cocos/spine/skeleton.ts [hunk 3]
```diff
@@ -227,10 +227,13 @@ export class Skeleton extends UIRenderer {
...
     protected _debugBones = false;
     @serializable
+    @editorOnly
```
Comment (bofeng-song) [inline cocos/spine/skeleton.ts]: It's not a editorOnly property, it can be work on native and other platforms

**Comment by yanjifa** [comment]
@bofeng-song please review again

File: cocos/spine/skeleton.ts [hunk 4]
```diff
@@ -1569,20 +1569,23 @@ export class Skeleton extends UIRenderer {
         if (this.debugBones || this.debugSlots || this.debugMesh) {
             if (!this._debugRenderer) {
                 const debugDrawNode = new Node('DEBUG_DRAW_NODE');
+                debugDrawNode.layer = this.node.layer;
                 debugDrawNode.hideFlags |= CCObject.Flags.DontSave | CCObject.Flags.HideInHierarchy;
                 const debugDraw = debugDrawNode.addComponent(Graphics);
-                debugDraw.lineWidth = 1;
+                debugDraw.lineWidth = 5;
```
Comment (bofeng-song) [inline cocos/spine/skeleton.ts:1575]: The modification of lineWidth is based on what considerations?

Comment (yanjifa) [reply cocos/spine/skeleton.ts:1575]: In cocos/spine/assembler/simple.ts, the lineWidth is consistently set to '5'. However, I have a need to adjust the lineWidth according to the worldScale of the animation. With this change, the engine's logic remains intact, and it allows me to define the linewidth within my code.

**Review by bofeng-song — APPROVED** [review]

**Comment by bofeng-song** [comment]
@cocos-robot run test cases

**Pull request merged**